src/runtime_src/core/common/info_aie.cpp

```diff
@@ -0,0 +1,692 @@
+                ],
+                "pc": [
+                    "0x12345678"
+                ],
+                "lr": [
+                    "0x45678901"
+                ],
+                "sp": [
+                    "0x78901234"
+                ]
+            },
+            "dma": {
+                "channel_status": {
+                    "mm2s": [
+                        "Running"
+                    ],
+                    "s2mm": [
+                        "Stalled_on_lock"
+                    ]
+                },
+                "queue_size": {
+                    "mm2s": [
+                        "2"
+                    ],
+                    "s2mm": [
+                        "3"
+                    ]
+                },
+                "queue_status": {
+                    "mm2s": [
+                        "channel0_overflow"
+                    ],
+                    "s2mm": [
+                        "channel0_overflow"
+                    ]
+                },
+                "current_bd": {
+                    "mm2s": [
+                        "3"
+                    ],
+                    "s2mm": [
+                        "2"
+                    ]
+                },
+            },
+            "lock": {
+                "lock0": [
+                    "Acquired_for_read"
+                ],
+                "lock1": [
+                    "Acquired_for_write"
+                ]
+            },
+            "errors": {
+                "core": {
+                    "Bus": [
+                        "AXI-MM_slave_error"
+                    ]
+                },
+                "memory": {
+                    "ECC": [
+                        "DM_ECC_error_scrub_2-bit",
+                        "DM_ECC_error_2-bit"
+                    ]
+                },
+                "pl": {
+                    "DMA": [
+                        "DMA_S2MM_0_error",
+                        "DMA_MM2S_1_error"
+                    ]
+                }
+            },
+            "event": {
+                "core": [
+                    "Perf_Cnt0",
+                    "PC_0",
+                    "Memory_Stall"
+                ],
+                "memory": [
+                    "Lock_0_Acquired",
+                    "DMA_S2MM_0_go_to_idle"
+                ],
+                "pl": [
+                    "DMA_S2MM_0_Error",
+                    "Lock_0_Acquired"
+                ]
+            }
+        },
+        ....
+*/
+
+boost::property_tree::ptree
+populate_aie_shim(const xrt_core::device * device, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  pt.put("description", desc);
+  boost::property_tree::ptree _pt;
+
+  try {
+  std::string aie_data = xrt_core::device_query<qr::aie_shim_info>(device);
+    std::stringstream ss(aie_data);
+    boost::property_tree::read_json(ss, _pt);
+  } catch (const std::exception& ex){
+    pt.put("error_msg", ex.what());
+    return pt;
+  }
+
```

Comment:
This function is too long and has no comments.

Reply:
Divided this function with sub functions. Please review once again